server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -253,6 +256,16 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
      */
     public static final String VIRTUAL_DATA_BLOB_PREFIX = "v__";
 
+    public static final String SNAPSHOT_REPOSITORY_DATA_CACHET_THRESHOLD_SETTING_NAME = "snapshot.repository_data.cache.threshold";
+
+    public static final double SNAPSHOT_REPOSITORY_DATA_CACHE_THRESHOLD_DEFAULT_PERCENTAGE = 0.01;
+
+    public static final long CACHE_MIN_THRESHOLD = ByteSizeUnit.KB.toBytes(500);
+
+    public static final long CACHE_MAX_THRESHOLD = calculateMaxSnapshotRepositoryDataCacheThreshold();
+
+    private static final int MAX_SAFE_ARRAY_SIZE = Integer.MAX_VALUE - 8;
```

Comment:
Don't do another commit just for this, but a link to the [JDK source code reference of this same constant](https://github.com/openjdk/jdk11u/blob/cee8535a9d3de8558b4b5028d68e397e508bef71/src/jdk.zipfs/share/classes/jdk/nio/zipfs/ByteArrayChannel.java#L226) in a comment would be helpful for future maintainers of this code.

Interestingly, that constant is defined multiple places. [This one is public](https://github.com/openjdk/jdk/blob/619b4d596634ee84b9bf5884b97a69eb01661657/src/java.base/share/classes/jdk/internal/util/ArraysSupport.java#L854) but of course in an "internal" package (go figure).

Reply:
I needed to make some code modifications, so I went ahead and added this comment as well :) Thank you!